### PR TITLE
review: test: add tests for CtCommentIMmpl class

### PR DESCRIPTION
**Pull request by Rohitesh-Kumar-Jain**
#1818

Hi, I have added some new tests for methods of the JavaOutputProcessor class :

These tests are intended to kill this mutation:

Screenshot 2021-07-18 at 3 02 08 PM

Link to #3967

### CODE REVIEW

File: src/test/java/spoon/test/comment/CommentTest.java [hunk 1]
```diff
@@ -775,6 +776,78 @@ public void testSnippedWithComments() {
...
+	@Test
+	public void testEqualsWithSameContentAndDifferentType() {
+		// contract: equals return false when comments with same content but different types are compared
+
+		Factory factory = new Launcher().getFactory();
+
+		CtComment comment = factory.Core().createComment();
+		String commentContent = "testContent";
+		comment.setCommentType(CtComment.CommentType.BLOCK);
+		comment.setContent(commentContent);
```
Comment (slarse) [inline src/test/java/spoon/test/comment/CommentTest.java:840]: Same comment about the names.

Can be shortened like so:

```suggestion
		CtComment comment = factory.createComment("testContent", CtComment.CommentType.BLOCK);
```

File: src/test/java/spoon/test/comment/CommentTest.java [hunk 2]
```diff
@@ -775,6 +776,78 @@ public void testSnippedWithComments() {
...
+
+		CtCommentImpl ctComment = new CtCommentImpl();
+		ctComment.setCommentType(CtComment.CommentType.INLINE);
+		ctComment.setContent(commentContent);
+
+		boolean expectedBoolean = ctComment.equals(comment);
```
Comment (slarse) [inline src/test/java/spoon/test/comment/CommentTest.java:846]: Not the expected value.

**Review by slarse — CHANGES_REQUESTED** [review]
Hi @Rohitesh-Kumar-Jain,

The test intents are good, but they can be made significantly shorter and even more clear with some slight tweaks.

File: src/test/java/spoon/test/comment/CommentTest.java [hunk 3]
```diff
@@ -775,6 +776,78 @@ public void testSnippedWithComments() {
...
+
+		Factory factory = new Launcher().getFactory();
+
+		CtComment comment = factory.Core().createComment();
+		String commentContent = "testContent";
+		comment.setContent(commentContent);
+
+		CtCommentImpl ctComment = new CtCommentImpl();
+		String ctCommentContent = "notTestContent";
+		ctComment.setContent(ctCommentContent);
```
[the remark this answers is not in the document]
Comment (Rohitesh-Kumar-Jain) [reply src/test/java/spoon/test/comment/CommentTest.java:804]: Both of them can be created from the factory, but I created one with CtCommentImpl(), as to emphasize that we are going to check `CtCommentImpl.equals` 😅

File: src/test/java/spoon/test/comment/CommentTest.java [hunk 4]
```diff
@@ -775,6 +776,78 @@ public void testSnippedWithComments() {
...
+
+		boolean expectedBoolean = ctComment.equals(comment);
```
[the remark this answers is not in the document]
Comment (Rohitesh-Kumar-Jain) [reply src/test/java/spoon/test/comment/CommentTest.java:801]: yes will change that as well

**Comment by Rohitesh-Kumar-Jain** [comment]
Other methods in this class don't need improvement in tests so, there's no hurry to merge this one

**Review by slarse — APPROVED** [review]
LGTM, thanks @Rohitesh-Kumar-Jain

**Pull request merged**